### ISSUE TITLE
[202405][DNX] orchagent exited because of failing to set SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL

### ISSUE THREAD

**Issue by ysmanman** [sonic-net/sonic-buildimage#19592]
#### Description

We noticed following orchagent failure in T2 testing with 202405 image.

```
2024 Jul 13 13:59:29.646591 xxx405-3 INFO swss#supervisord: orchagent
2024 Jul 13 13:59:40.815226 xxx405-3 ERR syncd#syncd: [none] SAI_API_NEIGHBOR:brcm_sai_set_neighbor_entry_attribute:597 Error processing nbr entry attribute failed with error Unknown error (0xfffd0000).
2024 Jul 13 13:59:40.815226 xxx405-3 ERR syncd#syncd: :- sendApiResponse: api SAI_COMMON_API_SET failed in syncd mode: SAI_STATUS_FAILURE
2024 Jul 13 13:59:40.815383 xxx405-3 ERR syncd#syncd: :- processQuadEvent: attr: SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL: false
2024 Jul 13 13:59:40.815875 xxx405-3 INFO swss#supervisord: orchagent
2024 Jul 13 13:59:40.815875 xxx405-3 ERR swss#orchagent: :- set: set status: SAI_STATUS_FAILURE
2024 Jul 13 13:59:40.815875 xxx405-3 ERR swss#orchagent: :- addNeighbor: Failed to update neighbor 00:11:22:33:44:55 on nfc405-7|Asic0|Ethernet180, attr.id=0x8, rv:-1
2024 Jul 13 13:59:40.815875 xxx405-3 ERR swss#orchagent: :- handleSaiSetStatus: Encountered failure in set operation, exiting orchagent, SAI API: SAI_API_NEIGHBOR, status: SAI_STATUS_FAILURE
```

The failure was observed with `arp/test_neighbor_mac_noptf.py` and `arp/test_arpall.py`.

#### Steps to reproduce the issue:
1.
2.
3.

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by ysmanman** [comment]
@arlakshm @kenneth-arista

**Comment by ysmanman** [comment]
Looked at BRCM SAI code and it seems neither 202205 nor 202405  SAI supports setting `SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL` for neighbor entry. But we didn't see the crash in 202205 testing. Maybe SONiC behaviors in 202205 and 202405 are different.

**Comment by ysmanman** [comment]
FYI, CSP CS00012298563 confirmed BRCM SAI did not support setting `SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL`  in SAI 9.x (or maybe earlier version too).

**Comment by arlakshm** [comment]
Thanks @ysmanman  for reporting this issue. @saksarav-nokia, @mlok-nokia  for viz..

**Comment by arlakshm** [comment]
@ysmanman, I did a quick check on the SAI definition. This attribute supports create and set. Any reason why the SAI behavior was changed?
![image](https://github.com/user-attachments/assets/97c5e58a-4b90-4684-9e29-56cfe140a302)

https://github.com/opencomputeproject/SAI/blob/dff0e34511e9a9018ee81743c95015f41a3f8c47/inc/saineighbor.h#L144-L156

**Comment by ysmanman** [comment]
Hi @arlakshm ,  I don't have too much context on why BRCM discontinued supporting setting `SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL` at least starting from SAI 9.2. But based on the conversion in CSP CS00012298563,  there were some discussion between MSFT and BRCM as well.  Quote the reply from BRCM:  

```
update:

at the meeting with MSFT, they are asking if SAI9.x supports SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL on brcm_sai_set_neighbor_entry_attribute() .

Answer: it is not supported on SAI 9.x```

**Comment by ysmanman** [comment]
Opened CSP CS00012360402 to track the issue.

**Comment by kenneth-arista** [comment]
The SONiC behavior changed between 202205 and 202405. Specifically, https://github.com/sonic-net/sonic-swss/pull/2577 fixed applying all neighbor attributes, which exposed this problem in the DNX SAI.

**Comment by kenneth-arista** [comment]
The fix from Broadcom is available in DNX SAI 11.2.7.1
